Automattic/jetpack
# Forms: Fix response details date value

## Proposed changes:

* Fix the Form response date value in the response details section
* Add a link to the source value in the response details section

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

* Make sure you have the Forms package enabled
* Go to the response inbox in the ` wp-admin`
* Select a response and check if the issues mentioned above were fixed in the response details section
